pkg/sql/plan2/function/function_id.go

```diff
@@ -0,0 +1,24 @@
+// Copyright 2021 - 2022 Matrix Origin
```

Comment:
Why do you need this file, I don't quite understand.

Reply:
here's what I'm thinking about.

You can found that I make a function GetFunctionByName, it will search the register and do type-check for each function we found. 
The GetFunctionByName will be called when we a make the plan node. finally generate the Expr_c expression.
But when we execute the plan, if we do the search work again, it's really foolish. So if we set a unique function id for every overload. we can jump the type-check work. 

Maybe we can just record the index of function, it can do the same work.
I will make a consider if we should remove this file. But I think a unique id to diff other overloads is important.

Reply:
I am using index to diff different overloads but not function_id now.